packages/dd-trace/src/profiling/profiler.js

```diff
@@ -82,6 +86,9 @@ class Profiler extends EventEmitter {
         this._logger.debug(`Started ${profiler.type} profiler in ${threadNamePrefix} thread`)
       }
 
+      this._spanFinishListener = this._onSpanFinish.bind(this)
```

Comment:
Should we enable endpoint count when endpoint collection is enabled ?
Also the similiraties of the code make make wonder if this does not belong to wall profiler.

Reply:
Yeah, there are similarities, but then I'd also need to somehow communicate the data from the wall profiler to here, maybe by generalizing the return value from "profile ()". I was thinking about it and concluded that endpoint counting can be a general higher-level profiler concern – it's unlikely any particular profile provider would return it differently. You are right that this should be gated by whether endpoint collection is enabled. I might also extract the web span utilities into a separate file so we don't repeat ourselves.

Reply:
Okay, extracted commonalities into a separate file.